### DISCUSSION TITLE
Problema, passthrough

### DISCUSSION THREAD

**Discussion by Francesco-Rugg** [arkypita/LaserGRBL#1416] [Q&A]
Ho un problema con la mia macchina, praticamente impostando una foto in modalità "passthrough" In fase di lavorazione il laser salta qualche linea, allego due foto, quella con più linee saltate è a 120 dpi mentre quella con meno linee è a 125 DPI, cosa posso fare? 
Ho provato a cambiare velocità, sia lenta che veloce, nessun cambiamento, ho cambiato anche il valore $120-121. Non riesco a capire quale potrebbe essere il problema. 

La mia macchina è una Openbuild con scheda grbl MKS dlc 2.0 settata su un microstepping di 1/16

Allego Foto

120 DPI

![IMG_20210415_095841.jpg](https://user-images.githubusercontent.com/62062196/114836282-c775a100-9dd2-11eb-9bda-681da926e3d9.jpg)

125 DPI

![IMG_20210415_095903.jpg](https://user-images.githubusercontent.com/62062196/114836354-d65c5380-9dd2-11eb-80dd-91d0ca9e96ba.jpg)

**Comment by gmmanonymus111** [comment]
I've made 2 test with passthrough, but mine has no noticable lines (bolt-driven engraver).
![image](https://user-images.githubusercontent.com/9658243/114868613-66f85b00-9df6-11eb-8752-f7a65cf58c3d.png)

Upper one is 200 DPI, the darker one 300 DPI.
If your machine is belt-driven, it might be the calculated steps sometimes get rounded, thus leaving empty lines. You could try using higher DPI resolution (300 is printing standard AFAIK), or one that is compatible with your stepping resolution (`$100` & `$101`).

**Comment by arkypita** [comment]
![image](https://user-images.githubusercontent.com/8782035/114871075-30700f80-9df9-11eb-999b-97411b667e78.png)

All those waves in your engraving make me think that you have improper tensioned belt.



You can set a resolution that is multiple of your step, by using the resolution helper.

![image](https://user-images.githubusercontent.com/8782035/114870878-f7379f80-9df8-11eb-8dc0-fcfcad368a34.png)

**Comment by Francesco-Rugg** [comment]
@arkypita Hi, thanks you for the fast response. 
With line to line option I don't have the problem of skipping line, but only if I use Passthrough. 

This is the photo with 130 DPI Line to line mode with 5.000 line/mm

![IMG_20210415_224330.jpg](https://user-images.githubusercontent.com/62062196/114937203-ccbc0580-9e3d-11eb-8391-c3119ff66349.jpg)

**Comment by arkypita** [comment]
What is your $100/101?

**Comment by Francesco-Rugg** [comment]
@arkypita 

$100=80.006 (x, step/mm)
$101=80.006 (y, step/mm)

I had calculated this number to fine tuning the machine

**Comment by Francesco-Rugg** [comment]
@arkypita 

$100=80.006 (x, step/mm)
$101=80.006 (y, step/mm)

I had calculated this number to fine tuning the machine


EDIT in the image that I posted in the comment before there is a chain that makes the shadow on the photo. 

This is a better one

![IMG_20210415_180155.jpg](https://user-images.githubusercontent.com/62062196/114941778-df393d80-9e43-11eb-8390-19309cd464ee.jpg)

**Comment by arkypita** [comment]
When you use line2line the original image DPI is not used. Image is scaled up/down to match the output size at the target resolution (as line/mm). The distance between two lines in this case is the inverse of the resolution.

i.e 5.000  -> 1/5 = 0.2mm

0.2 is a reliable resolution, as it is a (least common) multiple of your $100 (80 => 1/80 => 0,0125 and 0.2/16 = 0,0125) your motor make exactly 16 step between one line and next.

When using passthrough the original image DPI is used both for determine distance between two lines and image size.

120 dpi -> 25.4/120 = 0,2116666666666666666666666666667mm

A questo punto dovresti arrivarci da solo

**Comment by Francesco-Rugg** [comment]
@arkypita yes. 

Now, do I need to test another Control board?

**Comment by arkypita** [comment]
no, you need to use a DPI that your engraver is able to respect.
Any DPI that give a line spacing that is multiple of 0,0125mm (that is your motor step)

try 127 DPI

**Comment by Francesco-Rugg** [comment]
@arkypita ok, I understand. Tomorrow I'll do a test and let you know. But I think this solve the problem, thank you for the help

**Comment by arkypita** [comment]
Think of engraving a ruler and think your machine can only take 0.9mm steps
If you set 1 line / mm the gcode tells it to do: 0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10

But the machine cannot physically do them, so it will go to the approximately closest location. up to a certain point the nearest position will be the lower multiple of 0.9 but at some point the closest position will be the higher multiple.

At that moment instead of a jump of 0.9 there is a jump of 1.8

![image](https://user-images.githubusercontent.com/8782035/114944679-596bc100-9e48-11eb-892a-f967ff2272af.png)
